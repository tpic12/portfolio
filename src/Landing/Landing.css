.bold {
  font-weight: 700;
}

.landing-section {
  width: 100vw;
  height: 100vh;
  display: flex;
  /* flex-direction: column; */
  /* margin-top: 100px; */
  justify-content: center;
  align-items: flex-start;
  background-color: #ededed;
}

.landing-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  height: 75vh;
}

.landing-div {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 75vh;
  background-color: rgb(62, 72, 117);
  border-radius: 0 0 200px 200px;
}

.landing-div svg {
  position: relative;
  top: 130px;
  color: rgb(62, 72, 117);
  animation: bounce 3s ease-out;
  animation-iteration-count: infinite;
  align-self: center;
}

.landing-circle {
  border-radius: 50%;
  width: 200px;
  height: 200px;
  /* border: 3px solid black; */
  background-color: white;
  display: flex;
  justify-content: center;
  align-items: center;
}

.logo {
  font-size: 90px;
  font-weight: 500;
  color: rgb(62, 72, 117);
}

.name-tag-div {
  width: 75vw;
  color: white;
}

.name-tag-div h1 {
  font-size: 48px;
  margin: 1px;
}

.thin {
  font-weight: 400;
}

@keyframes bounce {
  0% {
    transform: translateY(0)
  }

  50% {
    transform: translateY(-80%)
  }

  100% {
    transform: translateY(0);
  }
}

@media screen and (min-width: 720px) {
  .landing-div {
    border-radius: 0;
    height: 60vh;
    padding: 0 20px;
  }

  .landing-content {
    flex-direction: row;
    height: 60vh;
    width: 80vw;
    align-self: center;
  }

  .landing-section {
    align-items: center;
  }

  .landing-circle {
    padding: 0 20px;
    margin: 0 10px;
  }

  .name-tag-div p {
    max-width: 380px;
  }

  .name-tag-div h1 {
    font-size: 60px;
  }
}

@media screen and (min-width: 950px) {
  .landing-content {
    max-width: 900px;
  }
}