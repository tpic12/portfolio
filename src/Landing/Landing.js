import React from "react";
import ProjectLinus from "../Linus/Linus";
import ProjectSpanimal from "../Spanimal/Spanimal";
import ProjectQtrail from "../Qtrail/Qtrail";
import About from "../About/About";
import Contact from "../Contact/Contact";
import ProjectLanding from "../ProjectLanding/ProjectLanding";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import "./Landing.css";

const Landing = ({ stickyRef }) => {
  const logo = () => {
    return "{tP}";
  };
  return (
    <>
      <section className="landing-section snap">
        <div className="landing-div" ref={stickyRef}>
          <div className="name-tag-div">
            <h1>
              <span className="thin">Taylor</span>
              <br />
              <span className="bold">Piccarreto</span>
            </h1>
            <p>
              Hi, Iâ€™m Taylor, and I like my applications like I like my
              pancakes: Fullstack and no bugs.
            </p>
          </div>
          <div className="landing-circle">
            <span className="logo">{logo()}</span>
          </div>
          <FontAwesomeIcon icon="chevron-down" size="lg" />
        </div>
      </section>
      <About />
      <div id="projects-section" className="projects snap">
        <ProjectLanding />
        <ProjectQtrail />
        <ProjectLinus />
        <ProjectSpanimal />
      </div>
      <Contact />
    </>
  );
};

export default Landing;
