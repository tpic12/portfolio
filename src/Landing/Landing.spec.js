import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Landing from "./Landing";

describe("<Landing />", () => {
  it("renders by default", () => {
    const wrapper = shallow(<Landing />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
