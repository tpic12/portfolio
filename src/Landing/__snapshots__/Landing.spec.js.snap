// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> renders by default 1`] = `
<Fragment>
  <section
    className="landing-section snap"
  >
    <div
      className="landing-div"
    >
      <div
        className="name-tag-div"
      >
        <h1>
          <span
            className="thin"
          >
            Taylor
          </span>
          <br />
          <span
            className="bold"
          >
            Piccarreto
          </span>
        </h1>
        <p>
          Hi, Iâ€™m Taylor, and I like my applications like I like my pancakes: Fullstack and no bugs.
        </p>
      </div>
      <div
        className="landing-circle"
      >
        <span
          className="logo"
        >
          {tP}
        </span>
      </div>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="chevron-down"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="lg"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
  </section>
  <About />
  <div
    className="projects snap"
    id="projects-section"
  >
    <ProjectLanding />
    <Qtrail />
    <Linus />
    <Spanimal />
  </div>
  <Contact />
</Fragment>
`;
